ipatests/test_webui/test_config.py

```diff
@@ -23,35 +23,490 @@
+
+        # test field with invalid value
+        self.assert_field_negative(usr_length_s, 'abc',
+                                   'Must be an integer')
+
+        # test field with exceeding value
+        self.assert_field_negative(usr_length_s, '9999',
+                                   'Maximum value is 255')
+
+        # test field with space in-between numbers
+        self.assert_field_negative(usr_length_s, '1 2',
+                                   'Must be an integer')
+
+        # test field with special char
+        self.assert_field_negative(usr_length_s, '*',
+                                   'Must be an integer')
+
+        # test if change of value is reflected
+        self.fill_input(usr_length_s, 3)
+        self.facet_button_click('save')
+        self.add_record(user_data.ENTITY, user_data.DATA, negative=True)
+        self.assert_last_error_dialog(ERR_MAX_CHAR.format(3))
+        self.close_all_dialogs()
+        self.navigate_to_entity(config_data.ENTITY)
+
+        # restore previous value
+        self.fill_input(usr_length_s, def_val)
+        self.facet_button_click('save')
+
+    @screenshot
+    def test_passwd_exp_notice(self):
+        """
+        Test "Password Expiration Notification" field
+        """
+        self.init_app()
+        self.navigate_to_entity(config_data.ENTITY)
+
+        exp_notice_s = 'ipapwdexpadvnotify'
+
+        # test field with blank field
+        self.assert_field_negative(exp_notice_s, '',
+                                   'Required field')
+
+        # test field with invalid value
+        self.assert_field_negative(exp_notice_s, 'abc',
+                                   'Must be an integer')
+
+        # test field with exceeding value
+        self.assert_field_negative(exp_notice_s, '9999999999',
+                                   'Maximum value is 2147483647')
+
+        # test field with space in-between numbers
+        self.assert_field_negative(exp_notice_s, '1 2',
+                                   'Must be an integer')
+
+        # test field with special char
+        self.assert_field_negative(exp_notice_s, '*',
+                                   'Must be an integer')
+
+    @screenshot
+    def test_group_search_field(self):
+        """
+        Test "Group search fields"
+        """
+        self.init_app()
+        self.navigate_to_entity(config_data.ENTITY)
+
+        group_search_s = 'ipagroupsearchfields'
+        def_val = self.get_field_value(group_search_s)
+
+        # test field with blank field
+        self.assert_field_negative(group_search_s, '',
+                                   'Required field')
+
+        # test field with invalid value
+        self.assert_field_negative(group_search_s, 'abc',
+                                   ERR_GRP_SEARCH_INV.format('"abc"'),
+                                   dialog=True)
+
+        # test field with leading space
+        self.assert_field_negative(group_search_s, ' leading_space',
```

Comment:
' leading_space' and bellow 'trailing_space ' is repeated so create variable for it (for reuse)